packages/web/src/common/store/buy-audio/slice.ts

```diff
@@ -0,0 +1,186 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { JupiterTokenSymbol } from 'common/store/buy-audio/constants'
+
+export enum Flow {
+  COINBASE_PAY = 'COINBASE_PAY'
+}
+
+export enum QuoteStatus {
+  IDLE = 'IDLE',
+  QUOTING = 'QUOTING',
+  SUCCEEDED = 'SUCCEEDED',
+  FAILED = 'FAILED'
+}
+
+export enum ExchangeStatus {
+  IDLE = 'IDLE',
+  WAITING = 'WAITING',
+  EXCHANGING = 'EXCHANGING',
+  SUCCEEDED = 'SUCCEEDED',
+  FAILED = 'FAILED'
+}
+
+type JupiterSwapActionPayload = {
+  inputTokenSymbol: JupiterTokenSymbol
+  outputTokenSymbol: JupiterTokenSymbol
+  inputAmount: number
+  forceFetch?: boolean
+  slippage?: number
+  padForSlippage?: boolean
+}
+
+type BuyAudioState = {
+  flow: Flow
+  swaps: Record<
+    JupiterTokenSymbol,
+    Record<
+      JupiterTokenSymbol,
+      {
+        inputUiAmount?: string
+        outputUiAmount?: string
+        quoteStatus: QuoteStatus
+        exchangeStatus: ExchangeStatus
```

Comment:
arguably we'll only ever support an exchange from SOL => AUDIO, so this could potentially be a top level status and `swaps` can be `quotes`

Reply:
did this. lmk what think